File: boj/11060.cpp
#include <iostream>
#include <vector>


using namespace std;

int n;

int main()
{
#ifdef DBG
    freopen("11060.txt","r",stdin);
#endif
    scanf("%d",&n);

    vector<int>v(n+10,0);
    vector<int>D(n+10,-1);

    D[0] = 0;


    for(int i = 0; i<n; i++){
        scanf("%d",&v[i]);
    }


    for(int i = 1; i<n; i++){
        for(int j = 0; j<i; j++){

            if(D[j] != -1 && i-j <= v[j]){

                if(D[i] == -1 || D[i] > D[j] +1){
                    D[i] = D[j] + 1;
                } 
            }
        }
    }


    cout<<D[n-1]<<endl;

    return 0;
}

File: boj/11060.py
import sys
dbg = 1 
if dbg: sys.stdin = open('in.txt','r')

n = int(input())
a = list(map(int, sys.stdin.readline().split()))
if dbg: print(a)
dp = [ -1 for i in range(n)]
dp[0] = 0

for i in range(1,n):
    j = 0
    while j < i:
        if dp[j] != -1 and i-j <= a[j]:
            if dbg: print(i,j)
            if dp[i] == -1 or dp[i] > dp[j] + 1:
                dp[i] = dp[j] + 1
        j += 1

print(dp[n-1])
if dbg: print(dp)
